src/components/Weather.js

```diff
@@ -39,7 +39,7 @@ class Weather extends React.Component {
     const weather = this.state.weather;
     let content = <div className='loading'>loading weather...</div>;
     
-    if (weather) {
+    if (weather?.main) {
```

Comment:
I think the question mark is a mistake?
```suggestion
    if (weather && weather.main) {
```